src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,601 @@
+        {
+            lastUpdateMs = nowMs;
+            observer_.onReceiveBitrateChanged(getSsrcs(), targetBitrateBps);
+        }
+    }
+
+    void timeoutStreams(long nowMs)
+    {
+        synchronized (critSect) {
+            Iterator<Map.Entry<Long, Long>> itr = ssrcs_.entrySet().iterator();
+            while (itr.hasNext()) {
+                Map.Entry<Long,Long> entry = itr.next();
+                if ((nowMs - entry.getValue() > kStreamTimeOutMs)) {
+                    itr.remove();
+                }
+            }
+            if (ssrcs_.isEmpty()) {
+                // We can't update the estimate if we don't have any active streams.
+                interArrival = new InterArrival((kTimestampGroupLengthMs
+                        << kInterArrivalShift) / 1000,
+                        kTimestampToMs, true);
+                estimator = new OveruseEstimator(new OverUseDetectorOptions());
+                // We deliberately don't reset the first_packet_time_ms_
+                // here for now since we only probe for bandwidth in the
+                // beginning of a call right now.
+            }
+        }
+    }
+
+    void OnRttUpdate(long avg_rtt_ms,
+                     long max_rtt_ms)
+    {
+        synchronized (critSect) {
+            remoteRate.setRtt(avg_rtt_ms);
+        }
+    }
+
+    void removeStream(long ssrc)
+    {
+        synchronized (critSect)
+        {
+            ssrcs_.remove(ssrc);
+        }
+    }
+
+    @Override
+    public long getLatestEstimate() {
+
+        synchronized (critSect)
+        {
+            long bitrateBps;
+            if (!remoteRate.isValidEstimate()) {
+                return -1;
+            }
+            if (ssrcs_.isEmpty()) {
+                bitrateBps = 0;
+            } else {
+                bitrateBps = remoteRate.getLatestEstimate();
+            }
+            return bitrateBps;
+        }
+    }
+
+    /**
+     * Gets the <tt>PacketTransformer</tt> for RTP packets.
+     *
+     * @return the <tt>PacketTransformer</tt> for RTP packets
+     */
+    @Override
+    public PacketTransformer getRTPTransformer() {
+        return this;
+    }
+
+    /**
+     * Gets the <tt>PacketTransformer</tt> for RTCP packets.
+     *
+     * @return the <tt>PacketTransformer</tt> for RTCP packets
+     */
+    @Override
+    public PacketTransformer getRTCPTransformer() {
+        return null;
+    }
+
+    /**
+     * Returns the estimated payload bitrate in bits per second if a valid
+     * estimate exists; otherwise, <tt>-1</tt>.
+     *
+     * @return the estimated payload bitrate in bits per seconds if a valid
+     * estimate exists; otherwise, <tt>-1</tt>
+     */
+
+    @Override
+    public Collection<Integer> getSsrcs() {
+
+        synchronized (critSect)
+        {
+            ssrcs = new ArrayList<>();
+            for(Long ssrcValue : ssrcs_.keySet()){
+                Number value = ssrcValue;
+                ssrcs.add(value.intValue());
```

Comment:
The only function of ssrc is to serve as a container for getting ssrc values anytime the getSsrc method is called, I declared it as a class member instead of inside the getSsrc method  to avoid re-declaring that variable anytime getSsrc() is called.

I have moved the declaration within the method. Though I was thinking there might be an added advantage or improvement by declaring once (as a class property) and assigning multiple times over declaring multiple times and assigning multiple times. Please correct me if am wrong. Thanks.